{PROJECT_NAME}/Sources/Data/NetworkAPI/Core/NetworkAPIProtocol.swift

```diff
@@ -12,33 +11,11 @@ protocol NetworkAPIProtocol {
 }
 
 extension NetworkAPIProtocol {
-
     func request<T: Decodable>(
-        session: Session,
-        configuration: RequestConfiguration,
-        decoder: JSONDecoder
+        provider: MoyaProvider<RequestConfiguration>,
```

Comment:
As I mentioned above, if I have `UserRequestConfiguation`, how can I make a request? This is not a core because it's only for the enum `RequestConfiguration`

Reply:
This's a good catch!

@Shayokh144 You might need to think about how we create different `RequestConfiguration`  (or `TargetType`) and handle it in the core service. On the other hand, it's better to show about testing code with Moya in the POW section

Reply:
So need a wrapper for the TargetType

`protocol RequestConfiguration: TargetType {}`

Reply:
@vnntsu @suho I have added a new commit [here](https://github.com/nimblehq/ios-templates/pull/394/commits/d37a1e0af77dc98bfedf7be31ded6a3ed901aee4). Now the core network api's are generic and reusable.